### DISCUSSION TITLE
Support postgis commands

### DISCUSSION THREAD

**Discussion by jo-chemla** [apache/superset#25054] [Ideas]
Root issue is https://github.com/apache/superset/issues/24861 : I've got a PostGIS DB and I'd like to run postgis commands on my rows properties - either through the SQL Lab within superset or when computing additional columns. 

Main use case is parsing columns which are WKB/WKT (polygons, polylines or points) to lat/lon floats/arrays without computing these fields on the db itself. Postgis could also be used to compute centroids, convex-hull, aggregate features, etc. Final use would then be to display the resulting geo-features on a map widget, either via mapbox, deckgl, openlayers etc - widgets like these already do exist within superset today, plus also see the [Generic Map Plugin](https://github.com/apache/superset/discussions/21758)

I was suggested by @rusackas to open an Ideas Dicsussion to discuss more this topic. 

As written in the original issue, I've tried enabling the postgis extension by replacing the postgres image with a postgis image in the docker-compose file, and could execute postgis commands within the container. But SqlLab won't let me execute the same commands, probably because sqlAlchemy does not recognize these commands as valid. Maybe using GeoAlchemy in place of SqlAlchemy could help, how simple would that be? Would there be any simpler way to execute postgis commands within superset? Thanks a lot for your feedback!

Similar discussions for supporting WKT or postgis geometries: https://github.com/apache/superset/discussions/18458 and https://github.com/apache/superset/discussions/18533

**Comment by rusackas** [comment]
I think @dpgaspar / @john-bodley / @betodealmeida / @villebro would be the ones most likely to know any difficulties/risks one would encounter when either replacing SqlAlchemy with GeoAlchemy. As a layperson here, I have only the most basic question - [why not both](https://github.com/apache/superset/assets/812905/e4fa508b-62b6-41cf-9e98-9363b794ec22)?

**Comment by betodealmeida** [comment]
> But SqlLab won't let me execute the same commands, probably because sqlAlchemy does not recognize these commands as valid.

This is not expected, the commands from SQL Lab are passed as is to the database, so they should work. Can you share the errors you get when doing this?

What would be nice is if the chart creation could leverage these extensions when creating maps, that would require some work. But just running the queries in SQL Lab should work fine.

**Comment by jakub-ga** [comment]
old issue, but I'm having exactly the same problem. I'm using Preset though.

here's the error. the same query works directly in postgreSQL with postgis installed
![image](https://github.com/user-attachments/assets/b0ff5d75-cc7d-499e-aa0d-f3c203ac47f4)
